### ISSUE TITLE
ENH: Command to force reinstall a package and all of its dependencies

### ISSUE THREAD

**Issue by dhirschfeld** [conda/conda#3149]

**Comment by sebma** [comment]
@kalefranz Hi,

Any news ?

**Comment by kalefranz** [comment]
This should be pretty easy to add now.  I added a 4.6 milestone to this.  No guarantees; it might slip.  But could also make it in.

**Comment by kalefranz** [comment]
The master branch right now has a `--force-reinstall` flag.  However that only applies the specs given on the command lines, and misses the "and all of its dependencies" part.  This issue is therefore still open.

**Comment by kalefranz** [comment]
This issue is related to https://github.com/conda/conda/issues/6247

**Comment by kalefranz** [comment]
I don't think this is going to make it into 4.6.  Like I said we do have a `--force-reinstall` flag.  Right now, the way that's implemented is the packages corresponding to what's given on the command line are all that's force-reinstalled.

We should actually look at how that flag can interact with `--no-deps` and `--only-deps`.  Basically, I think `--force-reinstall` should probably reinstall the command-line-given packages and all of their dependencies, then add in `--no-deps` or `--only-deps` to customize that behavior.

I guess that still leaves us needing some type of `--reinstall-all` or `--force-reinstall-all`, to cover #6247.
